### ISSUE TITLE
Atom causes whole system to freeze when changing a lot of files with git

### ISSUE THREAD

**Issue by StuntHacks** [atom/atom#18099]
### Description

When using git to change a bunch of files, Atom often causes the whole system to freeze, often up to the point where I have to hard-reset my computer.

### Steps to Reproduce

1. Open some files in a git repository
2. Use git to change a bunch of these files at once (`git checkout`, `git stash`, anything that affects multiple files)

**Expected behavior:** The system shouldn't freeze.

**Actual behavior:** The system freezes

**Reproduces how often:** ~80-90%

### Versions

1.30.0 x64 (non-snap)

### Additional Information

I'm on Ubuntu 18.04

**Comment by rsese** [comment]
Thanks for the report - it looks like the `Prerequisites` section of the issue template was removed, did you try [safe mode](https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode) or any of the other [debugging steps](https://flight-manual.atom.io/hacking-atom/sections/debugging)?

Also, do you see this behavior in any Git repository or just a particular one?

> Use git to change a bunch of these files at once (git checkout, git stash, anything that affects multiple files)

About how many files?  Just more than 1?

> 1.30.0 x64 (non-snap)

Just to confirm here, you installed using [the official .deb or the official package repository](https://flight-manual.atom.io/getting-started/sections/installing-atom/#debian-and-ubuntu-debapt)?

**Comment by StuntHacks** [comment]
> [...] did you try safe mode or any of the other debugging steps?
No, I didn't try safe mode yet. Since this issue is a bit "random", it's hard to know exactly when it will happen. But I'll try to reproduce it in safe mode.

> Also, do you see this behavior in any Git repository or just a particular one?
I've had the issue with multiple repositories.

> About how many files? Just more than 1?
I'm not sure what's the "minimum" amount is. It might happen with even just one file, but I never experienced that. In my experienced it happend with pretty much any amount of files, ranging from just 2 to about 10, though.

> Just to confirm here, you installed using the official .deb or the official package repository?
I installed it via the official package repository.

**Comment by rsese** [comment]
Great thanks :+1:

> I've had the issue with multiple repositories.

Are the repos particularly large?

Also, if you used Atom before 1.30, can you confirm if this behavior is new for you with 1.30 and things worked fine before?

**Comment by StuntHacks** [comment]
> Are the repos particularly large?

The one I'm working on right now, which caused me to report this issue, only has between 100 and 2000 files (depending on whether the documentation is built or not, it happens in both cases).

> Also, if you used Atom before 1.30, can you confirm if this behavior is new for you with 1.30 and things worked fine before?

It also happened on my old computer, which was running an earlier version of Atom. I'm not sure which one right now but I'll see and get back to this issue again!

**Comment by FacundoMainere** [comment]
this also happens to me
Debian 9
8gb ram, i5, SSD
Atom 1.29

**Comment by rsese** [comment]
> It also happened on my old computer, which was running an earlier version of Atom. I'm not sure which one right now but I'll see and get back to this issue again!

:+1:

Still not sure what the issue might be, just checking other potential issues - if you change `Settings > File System Watcher` to Experimental, does that make any difference?  Is there anything related to Atom in `/var/log/syslog`after you see this behavior?
